src/base/SoftEqual.hh

```diff
@@ -25,17 +25,13 @@ namespace celeritas
  * \param abs threshold for absolute error when comparing to zero
  *           (default 1.0e-14 for doubles)
  */
-template<typename T1 = real_type, typename T2 = T1>
+template<class T = ::celeritas::real_type>
```

Comment:
Is this intended only/most for struct and class or also for double and float.  If it is the later, `typename` seems more appropriate?

Reply:
Should just be for double and float. I've gone back and forth over the years about this -- recently my conviction changed toward "typename is used only for template aliasing, a la `using Foo = typename T::Foo`, and use `class` for template parameters" since there's nothing inherent to `typename` that makes it apply only to built-ins. It would be pretty easy, however, to change `T` to `RealType` as we do with the distributions.

Reply:
In this context, it can not be template aliasing, right?  So I see no advantage in distracting the user away from the fact that this class template expects a raw type rather than "appear" to be restricted to classes .... [and indeed there is no actual difference between the two spelling just the way they appear].

Reply:
You're correct that its meaning is unambiguous as `typename`, but the use of `typename` rather than `class` is inconsistent with the rest of the code, and indeed with the library definitions in the standard (which e.g. use `class RealType = double` for parameters that are meant for real or real-like types). So while I was changing the number of parameters to simplify, I thought I'd take the opportunity to standardize the use.

Reply:
> inconsistent with the rest of the code

Either this or the rest of the code could be changed :)

> and indeed with the library definitions in the standard

Fair enough.  (I am still have a slight preference for typename but not enough to worry).